hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc

```diff
@@ -0,0 +1,1570 @@
+# Define outgoing passthrough clock on the SPI_HOST_CLK pin but make sure it is a generated version
+# of the incoming passthrough clock, relate the host side pins to this clock in both input/output
+# directions.
+
+set SPI_DEV_FAST_PASS_TCK 25.0
+set SPI_DEV_FAST_PASS_TCK_HALF [expr ${SPI_DEV_FAST_PASS_TCK} / 2]
+create_clock -name SPI_DEV_FAST_PASS_CLK -period ${SPI_DEV_FAST_PASS_TCK} \
+    [get_ports ${SPI_DEV_CLK_PIN}] -add
+set_clock_uncertainty ${SETUP_CLOCK_UNCERTAINTY} [get_clocks SPI_DEV_FAST_PASS_CLK]
+
+# clocks used by spi device internally
+create_generated_clock -name SPI_DEV_FAST_PASS_IN_CLK \
+    -source [get_ports ${SPI_DEV_CLK_PIN}] -divide_by 1 \
+    -master_clock SPI_DEV_FAST_PASS_CLK -add \
+    [get_pins top_earlgrey/u_spi_device/u_clk_spi_in_buf/${CLK_PIN}]
+create_generated_clock -name SPI_DEV_FAST_PASS_OUT_CLK \
+    -source [get_ports ${SPI_DEV_CLK_PIN}] -divide_by 1 -invert \
+    -master_clock SPI_DEV_FAST_PASS_CLK -add \
+    [get_pins top_earlgrey/u_spi_device/u_clk_spi_out_buf/${CLK_PIN}]
+
+
+# clocks accounting for propagation delay to the other side
+create_generated_clock -name SPI_HOST_FAST_PASS_CLK \
+    -source [get_ports ${SPI_DEV_CLK_PIN}] -divide_by 1 \
+    -master_clock SPI_DEV_FAST_PASS_CLK -add \
+    [get_ports SPI_HOST_CLK]
+
+# The propagated properties are needed to ensure the passthrough clocks assume all passthrough delay.
+# This is done specifically for the passthrough interface to get realistic timing even during
+# pre-layout.
+set_propagated_clock [get_clock SPI_DEV_FAST_PASS_CLK]
+set_propagated_clock [get_clock SPI_HOST_FAST_PASS_CLK]
+
+# bidir ports facing host, with full-cycle sampling at the upstream host
+if {$spec_constr} {
+set_input_delay -min ${SPI_DEV_IN_DEL_MIN} ${SPI_DEV_DATA_PORTS} \
+    -clock_fall -clock SPI_DEV_FAST_PASS_CLK -add_delay
+set_input_delay -max ${SPI_DEV_IN_DEL_MAX} ${SPI_DEV_DATA_PORTS} \
+    -clock_fall -clock SPI_DEV_FAST_PASS_CLK -add_delay
+set_output_delay -min ${SPI_DEV_OUT_DEL_MIN_FC} ${SPI_DEV_DATA_PORTS} \
+    -clock_fall -clock SPI_DEV_FAST_PASS_CLK -add_delay
+set_output_delay -max ${SPI_DEV_OUT_DEL_MAX} ${SPI_DEV_DATA_PORTS} \
+    -clock_fall -clock SPI_DEV_FAST_PASS_CLK -add_delay
+} else {
+set_input_delay -min $spi_fast_pass_soc_in_min ${SPI_DEV_DATA_PORTS} \
+    -clock_fall -clock SPI_DEV_FAST_PASS_CLK -add_delay
+set_input_delay -max $spi_fast_pass_soc_in_max ${SPI_DEV_DATA_PORTS} \
+    -clock_fall -clock SPI_DEV_FAST_PASS_CLK -add_delay
+set_output_delay -min $spi_fast_pass_dev_out_val_min ${SPI_DEV_DATA_PORTS} \
+    -clock_fall -clock SPI_DEV_FAST_PASS_CLK -add_delay
+set_output_delay -max $spi_fast_pass_dev_out_val_max ${SPI_DEV_DATA_PORTS} \
+    -clock_fall -clock SPI_DEV_FAST_PASS_CLK -add_delay
+}
+set_multicycle_path -setup 2 -from [get_clocks SPI_DEV_FAST_PASS_IN_CLK] \
+    -to [get_clocks SPI_DEV_FAST_PASS_CLK] -through [get_ports ${SPI_DEV_DATA_PORTS}]
+#leonids  TBD need Alex approve
```

Comment:
Has this approval happened?

Reply:
The comment has been updated to say that this constraint got added based on interaction between Leonid and @a-will . This looks okay to me.